Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -152,29 +157,30 @@ public final class PackageInfoMapper: PackageInfoMapping {
     ///   - packageInfos: All available `PackageInfo`s
     ///   - productToPackage: Mapping from a product to its package
     ///   - packageToFolder: Mapping from a package name to its local folder
-    ///   - artifactsFolder: The folders containing downloaded SwiftPackageManager artifacts
+    ///   - packageToArtifacts: Mapping from a package name its artifacts' paths
     ///   - platforms: The configured platforms
     /// - Returns: Mapped project
     public func preprocess( // swiftlint:disable:this function_body_length
         packageInfos: [String: PackageInfo],
         productToPackage: [String: String],
         packageToFolder: [String: AbsolutePath],
-        artifactsFolder: AbsolutePath,
+        packageToArtifactPaths: [String: [AbsolutePath]],
         platforms: Set<TuistGraph.Platform>
     ) throws -> PreprocessInfo {
-        let targetDependencyToFramework: [String: Path] = packageInfos.reduce(into: [:]) { result, packageInfo in
-            let artifactsFolderForPackage = artifactsFolder.appending(component: packageInfo.key)
-            packageInfo.value.targets.forEach { target in
+        let targetDependencyToFramework: [String: Path] = try packageInfos.reduce(into: [:]) { result, packageInfo in
+            try packageInfo.value.targets.forEach { target in
                 guard target.type == .binary else { return }
                 if let path = target.path {
                     // local binary
                     result[target.name] = Path(packageToFolder[packageInfo.key]!.appending(RelativePath(path)).pathString)
                 } else {
-                    // remote binaries are checked out by SPM in artifacts/<Package>/<Target>.xcframework
-                    result[target.name] = Path(
-                        artifactsFolderForPackage.appending(component: "\(target.name).xcframework")
-                            .pathString
-                    )
+                    // remote binaries are checked out by SPM in artifacts/<Package.name>/<Target>.xcframework
+                    // or in artifacts/<Package.identity>/<Target>.xcframework when using SPM 5.6 and later
+                    guard let artifactPath = packageToArtifactPaths[packageInfo.key]?
+                        .first(where: { $0.components.last == "\(target.name).xcframework" }) else {
+                            throw PackageInfoMapperError.missingBinaryArtifact(package: packageInfo.key, target: target.name)
+                        }
+                    result[target.name] = Path(artifactPath.pathString)
```

Comment:
In the artifacts struct we are now decoding, we have the full mapping from package and target name to the xcframework, so I would use that instead of matching it by name, as it might not always work: [reference issue](https://github.com/tuist/tuist/issues/3718)

Reply:
Ok. I wasn't sure if the mapping between artifacts and targets is always 1-to-1 (I was wondering if there are some other artifact types that can be linked to a target, apart from XCFrameworks) so I chose the approach which was the most similar to the existing one. But if you think it's safe to make such assumption, then I'll go that way.

Reply:
@danyf90 What would be the best way to pass the collection of artifacts around? 
- `packageToTargetsToArtifactsPaths: [String: [String:  AbsolutePath]]`
- `packageToTargetArtifacts: [String: [TargetArtifact]]`, where
```
struct TargetArtifact {
    let targetName: String
    let path: AbsolutePath
}
```
- or something completely different?

Reply:
:sisi: only xcframeworks are supported AFAIK.
I think the `packageToTargetsToArtifactsPaths` is fine, or you can define a struct composed of package and target to be used as key of the dictionary

Reply:
Done